### PR TITLE
fix NoMethodError that occurs when generating scaffold inside full mode engine

### PR DESCRIPTION
**Pull request by y-yagi**
Currently, it is an error to generate the scaffold inside full mode engine. 

``` console
$ ./bin/rails -v  
Rails 4.2.3 

$ bin/rails g scaffold User name:string
      invoke  active_record
      create    db/migrate/20150628034727_create_users.rb
      create    app/models/user.rb
      invoke    test_unit
      create      test/models/user_test.rb
      create      test/fixtures/users.yml
      invoke  resource_route
       route    resources :users
      invoke  scaffold_controller
      create    app/controllers/users_controller.rb
      invoke    erb
      create      app/views/users
      create      app/views/users/index.html.erb
      create      app/views/users/edit.html.erb
      create      app/views/users/show.html.erb
      create      app/views/users/new.html.erb
      create      app/views/users/_form.html.erb
      invoke    test_unit
      create      test/controllers/users_controller_test.rb
/home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.3/lib/rails/generators/named_base.rb:98:in `namespaced_path': undefined method `name' for nil:NilClass (NoMethodError)
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.3/lib/rails/generators/test_unit/scaffold/scaffold_generator.rb:21:in `fixture_name'
    from (erb):6:in `block in template'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/file_manipulation.rb:305:in `call'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/file_manipulation.rb:305:in `block in capture'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/file_manipulation.rb:310:in `with_output_buffer'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/file_manipulation.rb:305:in `capture'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.3/lib/rails/generators/named_base.rb:43:in `module_namespacing'
    from (erb):3:in `template'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/2.2.0/erb.rb:863:in `eval'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/2.2.0/erb.rb:863:in `result'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/file_manipulation.rb:116:in `block in template'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:53:in `call'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:53:in `render'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:62:in `block (2 levels) in invoke!'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:62:in `open'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:62:in `block in invoke!'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/empty_directory.rb:116:in `call'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/empty_directory.rb:116:in `invoke_with_conflict_check'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:60:in `invoke!'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions.rb:94:in `action'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:25:in `create_file'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions/file_manipulation.rb:115:in `template'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.3/lib/rails/generators/named_base.rb:26:in `block in template'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.3/lib/rails/generators/named_base.rb:60:in `inside_template'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.3/lib/rails/generators/named_base.rb:25:in `template'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.3/lib/rails/generators/test_unit/scaffold/scaffold_generator.rb:14:in `create_test_files'
    from /home/yaginuma/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run' 
```

This bug was introduced in #20387.  It was my fault, sorry about this. 

This commit fixes error, and adds a test for generated test files.

### CODE REVIEW

File: railties/lib/rails/generators/test_unit/scaffold/scaffold_generator.rb [hunk 1]
```diff
@@ -21,13 +21,17 @@ def create_test_files
...
         @fixture_name ||=
-          if defined?(ENGINE_ROOT)
+          if mountable_engine?
             "%s_%s" % [namespaced_path, table_name]
           else
             table_name
           end
       end
 
+      def mountable?
```
Comment (egilburg) [inline railties/lib/rails/generators/test_unit/scaffold/scaffold_generator.rb]: Is this method needed in addition to `mountable_engine?`

Comment (y-yagi) [inline railties/lib/rails/generators/test_unit/scaffold/scaffold_generator.rb]: It is a corresponding mistake. Fixed. Thanks!

**Comment by senny** [comment]
@y-yagi thanks for fixing the bug, especially for adding regression coverage on running these tests :yellow_heart:!

**Comment by senny** [comment]
Backported to `4-2-stable` (https://github.com/rails/rails/commit/9360077e00aa2acea1824220025d01966aa795b1)

**Comment by y-yagi** [comment]
@senny Thanks for reviewing!

**Pull request merged**